packages/create-waku/src/tests/cli-with-args.test.ts

```diff
@@ -0,0 +1,200 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import type { SyncOptions, SyncResult } from 'execa';
+import { execaCommandSync } from 'execa';
+import { afterEach, beforeAll, describe, expect, test } from 'vitest';
+
+const CLI_PATH = path.join(__dirname, '../../dist/index.js');
+
+const projectName = 'test-waku-app';
+const genPath = path.join(__dirname, projectName);
+const genPathWithSubfolder = path.join(__dirname, '.test', projectName);
+
+const run = <SO extends SyncOptions>(
+  args: string[],
+  options?: SO,
+): SyncResult<SO> => {
+  // @ts-expect-error relies on exactOptionalPropertyTypes being false
+  return execaCommandSync(`node ${CLI_PATH} ${args.join(' ')}`, options);
```

Comment:
I wonder what this brings over `child_process.execSync`.

Reply:
from the docs it looks like it helps with escaping and different forms of IO. I don't think this use necessitates it, so I could remove it, but it does seem useful as a way to exec things with a little bit of help

https://www.npmjs.com/package/execa

Reply:
I had simply seen this in the clack/prompts test and copied their setup

Reply:
> I had simply seen this in the clack/prompts test and copied their setup

Ahh, I see.

I prefer execSync, that avoids mental overhead to read the code, if it works.
```suggestion
  return execSync(`node ${CLI_PATH} ${args.join(' ')}`, options);
```

Reply:
ah as I am reading now, the thing we get from this is the ease of checking `stdout` and setting options like cwd and timeout on the different `run`s. 

Changing to `execSync` would actually complicate the test running in this case quite a bit.

Reply:
Alright then.